# Joint State Controllers ---------------------------------------
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50  
# Trajectory Controllers ---------------------------------------
fullbody_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - LLEG_JOINT0
    - LLEG_JOINT1
    - LLEG_JOINT2
    - LLEG_JOINT3
    - LLEG_JOINT4
    - LLEG_JOINT5

    - RLEG_JOINT0
    - RLEG_JOINT1
    - RLEG_JOINT2
    - RLEG_JOINT3
    - RLEG_JOINT4
    - RLEG_JOINT5

    - LARM_JOINT0
    - LARM_JOINT1
    - LARM_JOINT2
    - LARM_JOINT3
    - LARM_JOINT4
    - LARM_JOINT5
    - LARM_JOINT6
    - LARM_JOINT7

    - RARM_JOINT0
    - RARM_JOINT1
    - RARM_JOINT2
    - RARM_JOINT3
    - RARM_JOINT4
    - RARM_JOINT5
    - RARM_JOINT6
    - RARM_JOINT7

    - CHEST_JOINT0
    - CHEST_JOINT1
    - CHEST_JOINT2

    - HEAD_JOINT0
    - HEAD_JOINT1
    
  gains: # Required because we're controlling an effort interface
    LLEG_JOINT0: {p: 10000, i: 0, d: 100}
    LLEG_JOINT1: {p: 10000, i: 0, d: 100}
    LLEG_JOINT2: {p: 10000, i: 0, d: 100}
    LLEG_JOINT3: {p: 1000, i: 0, d: 10}
    LLEG_JOINT4: {p: 100,  i: 0, d: 1}
    LLEG_JOINT5: {p: 100,  i: 0, d: 1}

    RLEG_JOINT0: {p: 10000, i: 0, d: 100}
    RLEG_JOINT1: {p: 10000, i: 0, d: 100}
    RLEG_JOINT2: {p: 10000, i: 0, d: 100}
    RLEG_JOINT3: {p: 1000, i: 0, d: 10}
    RLEG_JOINT4: {p: 100,  i: 0, d: 1}
    RLEG_JOINT5: {p: 100,  i: 0, d: 1}

    LARM_JOINT0: {p: 10000, i: 0, d: 100}
    LARM_JOINT1: {p: 10000, i: 0, d: 100}
    LARM_JOINT2: {p: 10000, i: 0, d: 100}
    LARM_JOINT3: {p: 10000, i: 0, d: 100}
    LARM_JOINT4: {p: 10000, i: 0, d: 100}
    LARM_JOINT5: {p: 1000,  i: 0, d: 10}
    LARM_JOINT6: {p: 1000,  i: 0, d: 10}
    LARM_JOINT7: {p: 1000,  i: 0, d: 10}
    
    RARM_JOINT0: {p: 10000, i: 0, d: 100}
    RARM_JOINT1: {p: 10000, i: 0, d: 100}
    RARM_JOINT2: {p: 10000, i: 0, d: 100}
    RARM_JOINT3: {p: 10000, i: 0, d: 100}
    RARM_JOINT4: {p: 10000, i: 0, d: 100}
    RARM_JOINT5: {p: 1000,  i: 0, d: 10}
    RARM_JOINT6: {p: 1000,  i: 0, d: 10}
    RARM_JOINT7: {p: 1000,  i: 0, d: 10}

    CHEST_JOINT0: {p: 10000, i: 0, d: 100}
    CHEST_JOINT1: {p: 10000, i: 0, d: 100}
    CHEST_JOINT2: {p: 10000, i: 0, d: 100}
    
    HEAD_JOINT0: {p: 100,  i: 0, d: 1}
    HEAD_JOINT1: {p: 100,  i: 0, d: 1}
    
  state_publish_rate:  100            # Override default
  action_monitor_rate: 100            # Override default
  stop_trajectory_duration: 0        # Override default

l_gripper_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - L_finger1_joint
  gains:
    L_finger1_joint: {p: 10,  i: 0, d: 1}
  state_publish_rate:  100            # Override default
  action_monitor_rate: 100            # Override default
  stop_trajectory_duration: 0        # Override default
  
r_gripper_controller:
  type: effort_controllers/JointTrajectoryController
  joints:
    - R_finger1_joint
  gains:
    R_finger1_joint: {p: 10,  i: 0, d: 1}
  state_publish_rate:  100            # Override default
  action_monitor_rate: 100            # Override default
  stop_trajectory_duration: 0        # Override default
  
# l_gripper_controller:
#   type: effort_controllers/JointPositionController
#   joint: L_finger1_joint
#   pid: {p: 10, i: 0, d: 1}

# r_gripper_controller:
#   type: effort_controllers/JointPositionController
#   joint: R_finger1_joint
#   pid: {p: 10, i: 0, d: 1}